applications/ParticleMechanicsApplication/custom_utilities/mpm_particle_generator_utility.h

```diff
@@ -234,6 +242,299 @@ namespace MPMParticleGeneratorUtility
                 }
             }
         }
+    }
+    /**
+     * @brief Function to Initiate material point condition.
+     * @details Generating particle condition using a designated shape functions
+     */
+
+    template<SizeType TDimension>
+    void GenerateMaterialPointCondition(ModelPart& rBackgroundGridModelPart,
```

Comment:
I would specialize with TDIM =2/3 and move the implementation to the CPP

Reply:
I moved the implementation to the cpp file